### ISSUE TITLE
Flaky test test_removenode_with_ignored_node

### ISSUE THREAD

**Issue by nyh** [scylladb/scylladb#17535]
The test `topology_experimental_raft`::`test_tablets_removenode`::`test_removenode_with_ignored_node` failed once in a debug run in CI:
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/testReport/junit/(root)/non-boost%20tests/topology_experimental_raft_test_tablets_removenode_debug_2/

The test failed to boot a cluster of five servers - `servers = await manager.servers_add(5, cmdline=cmdline)`

I couldn't figure out exactly what's wrong - is it just slowness and timeout - or a real bug?
One of the Scylla log files, https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2301.log, ends with the following suspicious message:

```
ERROR 2024-02-26 17:29:34,871 [shard 0:main] init - Startup failed: std::runtime_error (Bootstrap failed. See earlier errors (Rolled back: Failed to commit cdc generation: std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))))
```

@kbr-scylla maybe this looks familiar to you, or you can find more information from the logs?

**Comment by kbr-scylla** [comment]
I'll describe how I approach the investigation

Since it's a failure during server add, more details can be found inside test.py logs (not test case logs -- I mean the server-side test.py, where the cluster manager resides)

test.py logs here: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/test.py.debug-release-dev.log

And from jenkins failure page we know that we need to look for setup of `debug/topology_experimental_raft.test_tablets_removenode.2::test_removenode_with_ignored_node`

Here it is:
```
17:28:38.196 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Setting up debug/topology_experimental_raft.test_tablets_removenode.2::test_removenode_with_ignored_node
17:28:38.196 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Current cluster 8caf927e-d4bb-11ee-8b3b-16ffec0ccd0b is dirty after test debug/topology_experimental_raft.test_tablets_removenode.2::test_removenode_with_ignored_node, replacing with a new one...
17:28:38.196 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: 8caf927e-d4bb-11ee-8b3b-16ffec0ccd0b, running: ScyllaServer(2265, 127.245.68.32, e27fad8d-03dd-4289-9f9c-cee47d65d2e9), ScyllaServer(2264, 127.245.68.27, 8c9b240d-805f-46f2-a177-1f87a54ef391), ScyllaServer(2266, 127.245.68.24, 3542d9ce-2c18-49cf-9923-b7316216a2ce), stopped: ScyllaServer(2263, 127.245.68.8, 521d30f9-3ec1-43d5-9c78-cd83ddc3bd0a)) stopping
17:28:38.196 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopping ScyllaServer(2265, 127.245.68.32, e27fad8d-03dd-4289-9f9c-cee47d65d2e9) in scylla-2265
17:28:38.196 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopping ScyllaServer(2264, 127.245.68.27, 8c9b240d-805f-46f2-a177-1f87a54ef391) in scylla-2264
17:28:38.197 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopping ScyllaServer(2266, 127.245.68.24, 3542d9ce-2c18-49cf-9923-b7316216a2ce) in scylla-2266
17:28:38.422 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopped ScyllaServer(2264, 127.245.68.27, 8c9b240d-805f-46f2-a177-1f87a54ef391) in scylla-2264
17:28:38.423 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopped ScyllaServer(2265, 127.245.68.32, e27fad8d-03dd-4289-9f9c-cee47d65d2e9) in scylla-2265
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopped ScyllaServer(2266, 127.245.68.24, 3542d9ce-2c18-49cf-9923-b7316216a2ce) in scylla-2266
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: 8caf927e-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ScyllaServer(2263, 127.245.68.8, 521d30f9-3ec1-43d5-9c78-cd83ddc3bd0a), ScyllaServer(2265, 127.245.68.32, e27fad8d-03dd-4289-9f9c-cee47d65d2e9), ScyllaServer(2264, 127.245.68.27, 8c9b240d-805f-46f2-a177-1f87a54ef391), ScyllaServer(2266, 127.245.68.24, 3542d9ce-2c18-49cf-9923-b7316216a2ce)) releases ips {'127.245.68.8', '127.245.68.24', '127.245.68.32', '127.245.68.27'}
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Created new cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Created cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: )
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Got new Scylla cluster: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b
17:28:38.435 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Leasing Scylla cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) for test debug/topology_experimental_raft.test_tablets_removenode.2::test_removenode_with_ignored_node
17:28:38.435 INFO>  [26/Feb/2024:17:28:38 +0200] "PUT /cluster/before-test/test_removenode_with_ignored_node HTTP/1.1" 200 239 "-" "Python/3.11 aiohttp/3.9.3"
17:28:38.436 INFO>  [26/Feb/2024:17:28:38 +0200] "GET /cluster/running-servers HTTP/1.1" 200 159 "-" "Python/3.11 aiohttp/3.9.3"
17:28:38.438 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b waiting for new IP...
17:28:38.438 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b obtained new IP: 127.245.68.27
17:28:38.438 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) adding server...
17:28:38.438 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] installing Scylla server in /scylladir/testlog/x86_64/debug/scylla-2299...
17:28:38.441 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] starting server at host 127.245.68.27 in scylla-2299...
17:28:38.463 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b waiting for new IP...
17:28:38.464 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b obtained new IP: 127.245.68.32
17:28:38.464 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) adding server...
17:28:38.464 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] installing Scylla server in /scylladir/testlog/x86_64/debug/scylla-2300...
17:28:38.467 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] starting server at host 127.245.68.32 in scylla-2300...
17:28:38.489 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b waiting for new IP...
17:28:38.489 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b obtained new IP: 127.245.68.24
17:28:38.490 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) adding server...
17:28:38.490 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] installing Scylla server in /scylladir/testlog/x86_64/debug/scylla-2301...
17:28:38.493 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] starting server at host 127.245.68.24 in scylla-2301...
17:28:38.516 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b waiting for new IP...
17:28:38.516 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b obtained new IP: 127.245.68.8
17:28:38.517 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) adding server...
17:28:38.517 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] installing Scylla server in /scylladir/testlog/x86_64/debug/scylla-2302...
17:28:38.520 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] starting server at host 127.245.68.8 in scylla-2302...
17:28:38.543 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b waiting for new IP...
17:28:38.543 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b obtained new IP: 127.245.68.28
17:28:38.544 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] Cluster ScyllaCluster(name: b6b501da-d4bb-11ee-8b3b-16ffec0ccd0b, running: , stopped: ) adding server...
17:28:38.544 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] installing Scylla server in /scylladir/testlog/x86_64/debug/scylla-2303...
17:28:38.546 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] starting server at host 127.245.68.28 in scylla-2303...
```


so we tried to start the following servers:
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2299.log
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2300.log
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2301.log
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2302.log
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2303.log

We managed to start 4 out of 5:
```
17:28:43.628 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] started server at host 127.245.68.27 in scylla-2299, pid 294532
```

```
17:28:48.958 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] started server at host 127.245.68.8 in scylla-2302, pid 294538
```

```
17:28:55.431 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] started server at host 127.245.68.32 in scylla-2300, pid 294534
```

```
17:29:02.650 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] started server at host 127.245.68.28 in scylla-2303, pid 294540
```

but then
```
17:29:35.067 ERROR> [debug/topology_experimental_raft.test_tablets_removenode.2] failed to start the node, server_id 2301, IP 127.245.68.24, workdir scylla-2301, host_id a7f96c8b-e240-4d33-af7c-356e12107606, cql [not connected]
17:29:35.067 ERROR> [debug/topology_experimental_raft.test_tablets_removenode.2] last line of /scylladir/testlog/x86_64/debug/scylla-2301.log:
ERROR 2024-02-26 17:29:34,871 [shard 0:main] init - Startup failed: std::runtime_error (Bootstrap failed. See earlier errors (Rolled back: Failed to commit cdc generation: std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))))
17:29:35.067 INFO> [debug/topology_experimental_raft.test_tablets_removenode.2] stopping ScyllaServer(2301, 127.245.68.24, a7f96c8b-e240-4d33-af7c-356e12107606) in scylla-2301
17:29:35.067 ERROR> [debug/topology_experimental_raft.test_tablets_removenode.2] Failed to start Scylla server at host 127.245.68.24 in scylla-2301: failed to start the node, server_id 2301, IP 127.245.68.24, workdir scylla-2301, host_id a7f96c8b-e240-4d33-af7c-356e12107606, cql [not connected]
Check the log files:
/scylladir/testlog/x86_64/test.py.debug-release-dev.log
/scylladir/testlog/x86_64/debug/scylla-2301.log
17:29:35.068 ERROR> [debug/topology_experimental_raft.test_tablets_removenode.2] Exception when executing _cluster_servers_add: failed to start the node, server_id 2301, IP 127.245.68.24, workdir scylla-2301, host_id a7f96c8b-e240-4d33-af7c-356e12107606, cql [not connected]
Check the log files:
/scylladir/testlog/x86_64/test.py.debug-release-dev.log
/scylladir/testlog/x86_64/debug/scylla-2301.log
Traceback (most recent call last):
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 1143, in catching_handler
    ret = await handler(request)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 1300, in _cluster_servers_add
    s_infos = await self.cluster.add_servers(data.get('servers_num'), data.get('cmdline'), data.get('config'),
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 842, in add_servers
    return await asyncio.gather(*(self.add_server(None, cmdline, config, property_file, start, expected_error)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 816, in add_server
    await server.install_and_start(self.api, expected_error)
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 285, in install_and_start
    await self.start(api, expected_error)
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 499, in start
    report_error("failed to start the node")
  File "/jenkins/workspace/scylla-master/scylla-ci/scylla/test/pylib/scylla_cluster.py", line 485, in report_error
    raise RuntimeError(message + "\nCheck the log files:\n"
RuntimeError: failed to start the node, server_id 2301, IP 127.245.68.24, workdir scylla-2301, host_id a7f96c8b-e240-4d33-af7c-356e12107606, cql [not connected]
Check the log files:
/scylladir/testlog/x86_64/test.py.debug-release-dev.log
/scylladir/testlog/x86_64/debug/scylla-2301.log
```

Logs from topology coordinator may provide more details, need to find the coordinator node(s) -- most likely the first node that managed to start
Indeed in `scylla-2299.log` we have:
```
INFO  2024-02-26 17:28:41,636 [shard 0:strm] raft_topology - start topology coordinator fiber
```

and the failure:
```
INFO  2024-02-26 17:29:04,076 [shard 0:strm] raft_topology - entered `commit cdc generation` transition state
INFO  2024-02-26 17:29:04,076 [shard 0:strm] raft_topology - executing global topology command barrier, excluded nodes: {1e4ae431-f645-425a-aaf5-e0640e6b218a}
ERROR 2024-02-26 17:29:04,102 [shard 0:strm] raft_topology - transition_state::commit_cdc_generation, raft_topology_cmd::command::barrier failed, error std::runtime_error (raft topology: exec_global_command(barrier) failed with seastar::rpc::closed_error (connection is closed))
```

so coordinator tried to RPC all nodes (`exec_global_command`) but connection to some node got closed... Unfortunately it doesn't say to which one :/ (https://github.com/scylladb/scylladb/issues/16923) Maybe some node crashed which closed the connection?

We see failure detector marking one of the nodes as dead in the same second:
```
INFO  2024-02-26 17:29:04,319 [shard 0:main] raft_group_registry - marking Raft server 585560a9-e3b4-4916-9ac3-45cebd252de0 as dead for raft groups
```
perhaps this node crashed...

Aaaaand indeed
https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/7010/artifact/testlog/x86_64/debug/scylla-2302.log

```
ERROR 2024-02-26 17:29:03,435 [shard 1:main] database - regular column id 0 >= 0, at: 0xd3dd04a /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x5685f9c /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x5685cf7 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x5688075 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x5688acf /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x44bb40e /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x44bafc6 0xee41879 0xed4d863 0xe992b6a 0xe87e508 0xe6940c4 0xe4279a2 0xe4c365d 0xd62f576 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4715728 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x471dce8 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4721d80 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4821856 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481fa00 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481f8f0 /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481f05c /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x44c6518 /lib64/libc.so.6+0x8c946 /lib64/libc.so.6+0x11296f
   --------
   seastar::internal::coroutine_traits_base<void>::promise_type
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)> >(seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::noncopyable_function<seastar::future<void> (seastar::future<void>&&)>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
Aborting on shard 1.
Backtrace:
  0xd3dd04a
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x48b915c
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x48b8db3
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x46f2bc8
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x47238b5
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481eb59
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481ee9e
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481ecae
  /lib64/libc.so.6+0x3dbaf
  /lib64/libc.so.6+0x8e883
  /lib64/libc.so.6+0x3dafd
  /lib64/libc.so.6+0x2687e
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x44bafee
  0xee41879
  0xed4d863
  0xe992b6a
  0xe87e508
  0xe6940c4
  0xe4279a2
  0xe4c365d
  0xd62f576
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4715728
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x471dce8
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4721d80
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x4821856
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481fa00
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481f8f0
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x481f05c
  /jenkins/workspace/scylla-master/scylla-ci/scylla/build/debug/seastar/libseastar.so+0x44c6518
  /lib64/libc.so.6+0x8c946
  /lib64/libc.so.6+0x11296f
```

So, another occurrence of https://github.com/scylladb/scylladb/issues/16821 :)

**Comment by kbr-scylla** [comment]
Although this didn't happen during repair. But most likely it's the same root cause (non-isolated schema tables reads), fix is already on `next`: 0c376043ebe0d72364ed45763b0717aa4bd1955e

Closing as dup
